chain/pruned_block_dispatcher.go

```diff
@@ -0,0 +1,554 @@
+package chain
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/rand"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/peer"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/neutrino/query"
+	"github.com/lightningnetwork/lnd/ticker"
+)
+
+const (
+	// defaultRefreshPeersInterval represents the default polling interval
+	// at which we attempt to refresh the set of known peers.
+	defaultRefreshPeersInterval = 5 * time.Second
+
+	// defaultPeerReadyTimeout is the default amount of time we'll wait for
+	// a query peer to be ready to receive incoming block requests. Peers
+	// cannot respond to requests until the version exchange is completed
+	// upon connection establishment.
+	defaultPeerReadyTimeout = 15 * time.Second
+
+	// requiredServices are the requires services we require any candidate
+	// peers to signal such that we can retrieve pruned blocks from them.
+	requiredServices = wire.SFNodeNetwork | wire.SFNodeWitness
+
+	// prunedNodeService is the service bit signaled by pruned nodes on the
+	// network.
+	prunedNodeService wire.ServiceFlag = 2 << 10
```

Comment:
Ah yeah, we don't have this defined in `btcd` yet given we don't yet implement a pruned operating mode for it.